tt_metal/impl/buffers/circular_buffer.cpp

```diff
@@ -30,7 +30,7 @@ CircularBuffer::CircularBuffer(const CoreRangeSet &core_ranges, const CircularBu
     config_(config),
     locally_allocated_address_(std::nullopt) {
     if (this->config_.total_size() == 0) {
-        TT_THROW("Circular Buffer Config Error: Circular buffer size cannot be 0 B");
+        return;
```

Comment:
Will kernels just not interact with the CB at all if it's zero sized?
What's the use case instead of not creating the CB at all? For dynamic ops?

If this is a valid use case, can we just remove this if statement? I think we should still set the other properties ven if it's zero sized

Reply:
Yes, I think 0-sized CB will not be used at all. We're trying to add 0-volume tensor support everywhere, preferably without adding extra checks in every op for this. We already support creating regular Buffer of 0 size, which just does nothing.

Ok, I'll try removing this if altogether and run the tests.

Reply:
Okay sounds good. Ideally if it's going to be zero sized/unused, then ops shouldn't be creating the CB since there is dispatch/fw overhead for initializing, but if this is in a dynamic op where we always have the cb and just modify the size to avoid recompile, then seems valid to have it.